lib/src/native/realm_core.dart

```diff
@@ -1811,59 +1888,64 @@ class ConfigHandle extends HandleBase<realm_config> {
 }
 
 class RealmHandle extends HandleBase<shared_realm> {
+  int _counter = 0;
+
+  final Map<int, WeakReference<RootedHandleBase>> _children = {};
```

Comment:
I'm wondering why do we need a collection here `final Map<int, WeakReference<RootedHandleBase>> _children = {};`. Could you help me to understand?

Reply:
I see. You keep the handles to all the objects from a specific realm into the RealmHandle. Then you know what to release while releasing the realm.

Reply:
But doesn't this mean that:
- If I have a RealmOdject in a Realm;
- then I delete this RealmOdject;
- its handle will still be kept into RealmHandle
- RealmOdject's handle won't be released until the Realm is released.
Is that correct?

Reply:
We're keeping a map of weak references, so if your object goes out of scope and gets garbage collected, then its handle should be released whenever that happens. If it doesn't, then we'll release the handle whenever the Realm is closed.

Reply:
But is it going to be garbage collected if we keep its handle into a realm that is still not out of the scope?

Reply:
Why wouldn't it be? There's nothing keeping a reference to the object handle.

Reply:
Aren't they still in the `children ` collection in the `RealmHandle` of the Realm? It is just interesting to me. It won't be a problem since you are releasing the realm when the callback (`initialDataCallback` for example) completes.

Reply:
Those are weak references, so they should not be counted when the GC runs.